src/argilla/training/spacy.py

```diff
@@ -98,21 +99,45 @@ def __init__(
         self._eval_dataset_path = "./dev.spacy" if self._eval_dataset else None
 
         self.language = language or "en"
+        self.model = model
+
         self.gpu_id = gpu_id
+        self.use_gpu = False
         if self.gpu_id != -1:
+            self.use_gpu = spacy.prefer_gpu(self.gpu_id)
+            if self.use_gpu is False and self.model is not None:
+                self._logger.warn(
+                    "Since the GPU is not available the model will be ignored, "
+                    "and set to `None` so that the default CPU model will be used "
+                    "instead."
+                )
+                self.model = None
+
+        if self.use_gpu:
             try:
-                spacy.prefer_gpu(self.gpu_id)
+                require_version("spacy-transformers")
             except:
+                self._logger.warn(
+                    "`spacy-transformers` is not installed and it's required "
+                    "to use the GPU. Falling back to the CPU."
+                )
+            finally:
+                self.use_gpu = False
                 self.gpu_id = -1
 
         self.config = init_config(
             lang=self.language,
-            pipeline=[self._pipeline_name],
+            pipeline=self._pipeline,
+            optimize="efficiency",
+            gpu=self.use_gpu,
         )
         self.config["paths"]["train"] = self._train_dataset_path
         self.config["paths"]["dev"] = self._eval_dataset_path or self._train_dataset_path
-        self.config["paths"]["vectors"] = model
         self.config["system"]["seed"] = seed or 42
+        if self.use_gpu:
```

Comment:
I believe it makes more sense to default to certain models in a more explicit way. Also because we potentially do not only want to reuse the transformer/vectors but also other model components.

Reply:
Yes I agree, I'll be investigating this these days, so I'll move this back to draft since there are still some things I want/need to change before merging. Thanks!